tests/Analyze/TimeGen.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE LambdaCase #-}
+-- Hedgehog generators of times
+module Analyze.TimeGen where
+
+import           Hedgehog       hiding (Update)
+import qualified Hedgehog.Gen   as Gen
+import qualified Hedgehog.Range as Range
+import qualified Text.Printf    as T
+
+-- with a few problematic ones disabled
+data TimeFormatAtom
```

Comment:
It might be worth testing time formatting specifically, since these will be so sparsely covered in the current tests.